apps/web/src/screens/Profile/modals/ConfirmTransaction/index.tsx

```diff
@@ -78,7 +88,40 @@ const ConfirmTransactionScreen: FC<{
 			</Stack>
 
 			<Stack marginTop="auto" marginHorizontal="auto">
-				<NavBtn content="Continue" route="" />
+				<NavBtn
+					content="Continue"
+					route=""
+					onPress={async () => {
+						if (address) {
+							const transaction = await constructTransaction({
+								sender: address,
+								amount,
+								network: network.value as Networks,
+								receiver,
+								token: token.value as string,
+							});
+
+							const res = await requestSignAndSendTransaction(transaction);
+
+							if (res?.signatureString) {
+								modalActions.show({
+									id: 'transaction-successfully',
+									bindingDirection: BindDirections.InnerBottom,
+									component: TransactionSuccessfulScreen as never,
+									context: {
+										sender: address,
+										receiver,
+										token,
+										network,
+										amount,
+										signatureString: res.signatureString,
+									},
+								});
+								handleOnPressCloseBtn();
+							}
+						}
+					}}
```

Comment:
move this "logic" method to `Valtio` state level, better separate presentation layer versus logic layer